schematics: verify that generated code passes CLI's stricter compiler flags

#### Reproduction
PR https://github.com/angular/angular-cli/pull/14905 introduces a new `strict` option to the Angular CLI. This option is currently `false` by default, but there are plans to make it `true` in version 9 or 10.

```json
    "noImplicitAny": true,
    "noImplicitReturns": true,
    "noImplicitThis": true,
    "noFallthroughCasesInSwitch": true,
    "strictNullChecks": true,
```
 
#### Expected Behavior
We need to make sure that all of the code generated by our Schematics will compile with these new flags.

We already use these flags for building the [library source](https://github.com/angular/components/blob/master/tsconfig.json) and [examples](https://github.com/angular/components/blob/cf70c4342e7ea277011cf289e721383b7784a18a/src/material-examples/tsconfig-build.json).

PR https://github.com/angular/material.angular.io/pull/631 adds these compiler flags for material.angular.io builds as well.

#### Related Issues
I'm not sure if this is something that we can do in integration tests (https://github.com/angular/components/issues/16479) or not. But it seems possible and would be needed to avoid introducing regressions over time.

https://github.com/angular/components/issues/13329 tracks making the library code follow even stricter checks (`--strictPropertyInitialization`).

https://github.com/angular/components/issues/16028 relates to stricter template checks in Ivy.

I'm currently seeing the following compilation errors after enabling these stricter settings in a project that just has Angular Material Schematics in it.
```sh
ERROR in app/table/table-datasource.ts:107:18 - error TS7006: Parameter 'a' implicitly has an 'any' type.

107 function compare(a, b, isAsc) {
                     ~
app/table/table-datasource.ts:107:21 - error TS7006: Parameter 'b' implicitly has an 'any' type.

107 function compare(a, b, isAsc) {
                        ~
app/table/table-datasource.ts:107:24 - error TS7006: Parameter 'isAsc' implicitly has an 'any' type.

107 function compare(a, b, isAsc) {
                           ~~~~~
app/tree/tree.component.ts:46:7 - error TS2345: Argument of type '(node: FileNode) => Observable<FileNode[] | undefined>' is not assignable to parameter of type '(node: FileNode) => FileNode[] | Observable<FileNode[]> | null | undefined'.
  Type 'Observable<FileNode[] | undefined>' is not assignable to type 'FileNode[] | Observable<FileNode[]> | null | undefined'.
    Type 'Observable<FileNode[] | undefined>' is not assignable to type 'Observable<FileNode[]>'.
      Type 'FileNode[] | undefined' is not assignable to type 'FileNode[]'.
        Type 'undefined' is not assignable to type 'FileNode[]'.

46       this.getChildren);
         ~~~~~~~~~~~~~~~~

```

--- a/src/material/schematics/ng-generate/table/files/__path__/__name@dasherize@if-flat__/__name@dasherize__-datasource.ts.template
+++ b/src/material/schematics/ng-generate/table/files/__path__/__name@dasherize@if-flat__/__name@dasherize__-datasource.ts.template
@@ -103,6 +103,6 @@ export class <%= classify(name) %>DataSource extends DataSource<<%= classify(nam
 }
 
 /** Simple sort comparator for example ID/Name columns (for client-side sorting). */
-function compare(a, b, isAsc) {
+function compare(a: string | number, b: string | number, isAsc: boolean) {
   return (a < b ? -1 : 1) * (isAsc ? 1 : -1);
 }

--- a/src/material/schematics/ng-generate/tree/files/__path__/__name@dasherize@if-flat__/__name@dasherize__.component.ts.template
+++ b/src/material/schematics/ng-generate/tree/files/__path__/__name@dasherize@if-flat__/__name@dasherize__.component.ts.template
@@ -1,6 +1,5 @@
 import { Component<% if (!!viewEncapsulation) { %>, ViewEncapsulation<% }%><% if (changeDetection !== 'Default') { %>, ChangeDetectionStrategy<% }%> } from '@angular/core';
 import { MatTreeFlatDataSource, MatTreeFlattener } from '@angular/material/tree';
-import { of as observableOf } from 'rxjs';
 import { FlatTreeControl } from '@angular/cdk/tree';
 import { files } from './example-data';
 
@@ -84,7 +83,7 @@ export class <%= classify(name) %>Component {
   }
 
   /** Get the children for the node. */
-  getChildren(node: FileNode) {
-    return observableOf(node.children);
+  getChildren(node: FileNode): FileNode[] | null | undefined {
+    return node.children;
   }
 }

